### DISCUSSION TITLE
How to optimize complex queries and avoid N+1 issues in SQLAlchemy without compromising performance?

### DISCUSSION THREAD

**Discussion by gabrielmrts** [sqlalchemy/sqlalchemy#12092] [Usage Questions]
I’m struggling to optimize complex queries that involve multiple relationships and avoid N+1 query issues in SQLAlchemy. The complexity of these operations requires efficient handling, but I want to make sure that performance isn't sacrificed. Are there best practices or patterns in SQLAlchemy to manage this effectively, particularly with joined or subqueries? Any insights or advanced tips would be highly appreciated!

**Comment by zzzeek** [comment]
eager loading is the main way to deal with this see https://docs.sqlalchemy.org/en/20/orm/queryguide/relationships.html
